KDE Survey 20220209

eatradish opened the pull request:
<!-- For description on topic creation and maintenance, please refer to [this Wiki article](https://wiki.aosc.io/developer/packaging/topic-based-maintenance-guideline/). -->

Topic Description
-----------------

KDE Survey 20220209

Package(s) Affected
-------------------

KF5 5.91.0
Plasma 5.24.3
KDE Applications 21.12.3

Security Update?
----------------

<!-- If this topic contains security update(s), please uncomment "Yes,"
     mark with the `security` and `priority` labels, and make sure to mark your commits to relevant issue numbers.

     Please see GitHub's documentation on "Linking a pull request to an issue":

     https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue -->

No

<!-- Please uncomment the "Build Order" section if your topic affects more than one package. -->

Build Order
-----------

```
qt-5 wayland-protocols plasma-wayland-protocols libinput groups/kf5 groups/plasma groups/kde-applications groups/kde-survey-20220209-rebuilds
```

For architectures without QtWebEngine support...

```
qt-5 wayland-protocols plasma-wayland-protocols libinput groups/kf5 groups/plasma groups/kde-applications-no-webengine groups/kde-survey-20220209-rebuilds 
```

Test Build(s) Done
------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [x] AMD64 `amd64`   
- [x] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->    
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

**Secondary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [x] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`

<!-- Maintainers should review file changes and, if the build script(s) affected complies with the
     [Package Styling Manual](https://wiki.aosc.io/developer/packaging/package-styling-manual/), please
     add `lgtm` label to this issue. -->

<!-- Maintainers and users may now test the packages in this topic and, once user/maintainer feedback indicates that the update(s) work as expected and find its quality satisfactory,
     another maintainer may now review this pull request and mark it as Approved. After which, the maintainer will build affected package(s) and upload them to the `stable` repository. -->

Update(s) Uploaded to Stable
----------------------------

**Primary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

- [ ] AMD64 `amd64`   
- [ ] AArch64 `arm64`
<!-- If this package involves a `+32` counterpart, please uncomment the line below. -->
<!-- - [ ] 32-bit Optional Environment `optenv32` -->

<!-- If all package(s) affected by this topic is `noarch`, please use the stub below. -->
<!-- - [ ] Architecture-independent `noarch` -->

**Secondary Architectures**

<!-- Please remove any architecture to which this topic does not apply. -->

Architectural progress for "secondary," or experimental ports does not impede on merging of this topic.

- [ ] Loongson 3 `loongson3`
- [ ] PowerPC 64-bit (Little Endian) `ppc64el`

<!-- TODO: CI to auto-fill architectural progress. -->

eatradish committed:
kqtquickcharts: update to 21.12.3

eatradish committed:
libkcompactdisc: update to 21.12.3

eatradish committed:
kcalutils: update to 21.12.3

eatradish committed:
libkcddb: update to 21.12.3

eatradish committed:
kmime: update to 21.12.3

eatradish committed:
kpimtextedit: update to 21.12.3

eatradish committed:
libkeduvocdocument: update to 21.12.3

eatradish committed:
kidentitymanagement: update to 21.12.3

eatradish committed:
libksane: update to 21.12.3

eatradish committed:
analitza: update to 21.12.3

eatradish committed:
libkipi: update to 21.12.3

eatradish committed:
kipi-plugins: update to 21.12.3

eatradish committed:
libkdegames: update to 21.12.3

eatradish committed:
libkomparediff2: update to 21.12.3

eatradish committed:
kdevelop: update to 21.12.3

eatradish committed:
libkmahjongg: update to 21.12.3

eatradish committed:
kontactinterface: update to 21.12.3

eatradish committed:
akonadi-calendar: update to 21.12.3

eatradish committed:
kimap: update to 21.12.3

eatradish committed:
kmailtransport: update to 21.12.3

eatradish committed:
akonadi-search: update to 21.12.3

eatradish committed:
akonadi-contacts: update to 21.12.3

eatradish committed:
ksmtp: update to 21.12.3

eatradish committed:
mailimporter: update to 21.12.3

eatradish committed:
libkgapi: update to 21.12.3

eatradish committed:
akonadi: update to 21.12.3

eatradish committed:
libkdepim: update to 21.12.3

eatradish committed:
grantleetheme: update to 21.12.3

eatradish committed:
kmbox: update to 21.12.3

eatradish committed:
pimcommon: update to 21.12.3

eatradish committed:
akonadi-mime: update to 21.12.3

eatradish committed:
kalarmcal: update to 21.12.3

eatradish committed:
kldap: update to 21.12.3

eatradish committed:
ktnef: update to 21.12.3

eatradish committed:
akonadi-notes: update to 21.12.3

eatradish committed:
libkleo: update to 21.12.3

eatradish committed:
kaccounts-integration: update to 21.12.3

eatradish committed:
libgravatar: update to 21.12.3

eatradish committed:
kcron: update to 21.12.3

eatradish committed:
libktorrent: update to 21.12.3

eatradish committed:
khelpcenter: update to 21.12.3

eatradish committed:
akonadi-import-wizard: update to 21.12.3

eatradish committed:
ktp-common-internals: update to 21.12.3

eatradish committed:
kaccounts-providers: update to 21.12.3

eatradish committed:
ksystemlog: update to 21.12.3

eatradish committed:
kmouth: update to 21.12.3

eatradish committed:
kmag: update to 21.12.3

eatradish committed:
signon-kwallet-extension: update to 21.12.3

eatradish committed:
incidenceeditor: update to 21.12.3

eatradish committed:
mailcommon: update to 21.12.3

eatradish committed:
libksieve: update to 21.12.3

eatradish committed:
eventviews: update to 21.12.3

eatradish committed:
partitionmanager: update to 21.12.3

eatradish committed:
kdepim-runtime: update to 21.12.3

eatradish committed:
kmousetool: update to 21.12.3

eatradish committed:
kate: update to 21.12.3

eatradish committed:
kdialog: update to 21.12.3

eatradish committed:
messagelib: update to 21.12.3

eatradish committed:
calendarsupport: update to 21.12.3

eatradish committed:
kgeography: update to 21.12.3

eatradish committed:
konqueror: update to 21.12.3

eatradish committed:
khangman: update to 21.12.3

eatradish committed:
parley: update to 21.12.3

eatradish committed:
ktouch: update to 21.12.3

eatradish committed:
kturtle: update to 21.12.3

eatradish committed:
kwordquiz: update to 21.12.3

eatradish committed:
kalzium: update to 21.12.3

eatradish committed:
kiten: update to 21.12.3

eatradish committed:
kanagram: update to 21.12.3

eatradish committed:
kfind: update to 21.12.3

eatradish committed:
kalgebra: update to 21.12.3

eatradish committed:
kmplot: update to 21.12.3

eatradish committed:
klettres: update to 21.12.3

eatradish committed:
kbruch: update to 21.12.3

eatradish committed:
minuet: update to 21.12.3

eatradish committed:
konversation: update to 21.12.3

eatradish committed:
artikulate: update to 21.12.3

eatradish committed:
marble: update to 21.12.3

eatradish committed:
cantor: update to 21.12.3

eatradish committed:
kig: update to 21.12.3

eatradish committed:
blinken: update to 21.12.3

eatradish committed:
klickety: update to 21.12.3

eatradish committed:
kdiamond: update to 21.12.3

eatradish committed:
kbreakout: update to 21.12.3

eatradish committed:
bovo: update to 21.12.3

eatradish committed:
kblocks: update to 21.12.3

eatradish committed:
bomber: update to 21.12.3

eatradish committed:
klines: update to 21.12.3

eatradish committed:
rocs: update to 21.12.3

eatradish committed:
step: update to 21.12.3

eatradish committed:
granatier: update to 21.12.3

eatradish committed:
kjumpingcube: update to 21.12.3

eatradish committed:
kgoldrunner: update to 21.12.3

eatradish committed:
kigo: update to 21.12.3

eatradish committed:
kblackbox: update to 21.12.3

eatradish committed:
kiriki: update to 21.12.3

eatradish committed:
katomic: update to 21.12.3

eatradish committed:
kapman: update to 21.12.3

eatradish committed:
kfourinline: update to 21.12.3

eatradish committed:
killbots: update to 21.12.3

eatradish committed:
kbounce: update to 21.12.3

eatradish committed:
kajongg: update to 21.12.3

eatradish committed:
knavalbattle: update to 21.12.3

eatradish committed:
kpat: update to 21.12.3

eatradish committed:
kollision: update to 21.12.3

eatradish committed:
picmi: update to 21.12.3

eatradish committed:
knights: update to 21.12.3

eatradish committed:
kmahjongg: update to 21.12.3

eatradish committed:
ksquares: update to 21.12.3

eatradish committed:
lskat: update to 21.12.3

eatradish committed:
kmines: update to 21.12.3

eatradish committed:
knetwalk: update to 21.12.3

eatradish committed:
gwenview: update to 21.12.3

eatradish committed:
palapeli: update to 21.12.3

eatradish committed:
kubrick: update to 21.12.3

eatradish committed:
ksnakeduel: update to 21.12.3

eatradish committed:
kolf: update to 21.12.3

eatradish committed:
konquest: update to 21.12.3

eatradish committed:
kshisen: update to 21.12.3

eatradish committed:
ksirk: update to 21.12.3

eatradish committed:
kreversi: update to 21.12.3

eatradish committed:
ktuberling: update to 21.12.3

eatradish committed:
ksudoku: update to 21.12.3

eatradish committed:
kspaceduel: update to 21.12.3

eatradish committed:
audiocd-kio: update to 21.12.3

eatradish committed:
kolourpaint: update to 21.12.3

eatradish committed:
kdenlive: update to 21.12.3

eatradish committed:
svgpart: update to 21.12.3

eatradish committed:
kget: update to 21.12.3

eatradish committed:
skanlite: update to 21.12.3

eatradish committed:
juk: update to 21.12.3

eatradish committed:
ktorrent: update to 21.12.3

eatradish committed:
k3b: update to 21.12.3

eatradish committed:
ffmpegthumbs: update to 21.12.3

eatradish committed:
dragon: update to 21.12.3

eatradish committed:
spectacle: update to 21.12.3

eatradish committed:
okular: update to 21.12.3

eatradish committed:
kcolorchooser: update to 21.12.3

eatradish committed:
kwave: update to 21.12.3

eatradish committed:
krdc: update to 21.12.3

eatradish committed:
kamera: update to 21.12.3

eatradish committed:
kruler: update to 21.12.3

eatradish committed:
krfb: update to 21.12.3

eatradish committed:
kmix: update to 21.12.3

eatradish committed:
kdenetwork-filesharing: update to 21.12.3

eatradish committed:
ktp-contact-list: update to 21.12.3

eatradish committed:
ktp-auth-handler: update to 21.12.3

eatradish committed:
ktp-approver: update to 21.12.3

eatradish committed:
kitinerary: update to 21.12.3

eatradish committed:
ktp-call-ui: update to 21.12.3

eatradish committed:
ktp-accounts-kcm: update to 21.12.3

eatradish committed:
ktp-send-file: update to 21.12.3

eatradish committed:
itinerary: update to 21.12.3

eatradish committed:
akonadiconsole: update to 21.12.3

eatradish committed:
ktp-kded-module: update to 21.12.3

eatradish committed:
ktp-text-ui: update to 21.12.3

eatradish committed:
ktp-desktop-applets: update to 21.12.3

eatradish committed:
grantlee-editor: update to 21.12.3

eatradish committed:
kpublictransport: update to 21.12.3

eatradish committed:
akonadi-calendar-tools: update to 21.12.3

eatradish committed:
kalarm: update to 21.12.3

eatradish committed:
ktp-filetransfer-handler: update to 21.12.3

eatradish committed:
ktp-contact-runner: update to 21.12.3

eatradish committed:
zeroconf-ioslave: update to 21.12.3

eatradish committed:
kaddressbook: update to 21.12.3

eatradish committed:
kpkpass: update to 21.12.3

eatradish committed:
akregator: update to 21.12.3

eatradish committed:
kosmindoormap: update to 21.12.3

eatradish committed:
dolphin-plugins: update to 21.12.3

eatradish committed:
kcachegrind: update to 21.12.3

eatradish committed:
kmail: update to 21.12.3

eatradish committed:
kompare: update to 21.12.3

eatradish committed:
cervisia: update to 21.12.3

eatradish committed:
kontact: update to 21.12.3

eatradish committed:
kirigami-gallery: update to 21.12.3

eatradish committed:
lokalize: update to 21.12.3

eatradish committed:
kmail-account-wizard: update to 21.12.3

eatradish committed:
kleopatra: update to 21.12.3

eatradish committed:
kde-dev-utils: update to 21.12.3

eatradish committed:
kopeninghours: update to 21.12.3

eatradish committed:
kde-dev-scripts: update to 21.12.3

eatradish committed:
korganizer: update to 21.12.3

eatradish committed:
pim-sieve-editor: update to 21.12.3

eatradish committed:
kdepim-addons: update to 21.12.3

eatradish committed:
kapptemplate: update to 21.12.3

eatradish committed:
mbox-importer: update to 21.12.3

eatradish committed:
zanshin: update to 21.12.3

eatradish committed:
pim-data-exporter: update to 21.12.3

eatradish committed:
knotes: update to 21.12.3

eatradish committed:
kfloppy: update to 21.12.3

eatradish committed:
poxml: update to 21.12.3

eatradish committed:
print-manager: update to 21.12.3

eatradish committed:
elisa: update to 21.12.3

eatradish committed:
markdownpart: update to 21.12.3

eatradish committed:
sweeper: update to 21.12.3

eatradish committed:
kdf: update to 21.12.3

eatradish committed:
kwalletmanager: update to 21.12.3

eatradish committed:
ark: update to 21.12.3

eatradish committed:
ktimer: update to 21.12.3

eatradish committed:
filelight: update to 21.12.3

eatradish committed:
kamoso: update to 21.12.3

eatradish committed:
kopete: update to 21.12.3

eatradish committed:
kdebugsettings: update to 21.12.3

eatradish committed:
kcalc: update to 21.12.3

eatradish committed:
kimagemapeditor: update to 21.12.3

eatradish committed:
kteatime: update to 21.12.3

eatradish committed:
kcharselect: update to 21.12.3

eatradish committed:
yakuake: update to 21.12.3

eatradish committed:
umbrello: update to 21.12.3

eatradish committed:
kgpg: update to 21.12.3

eatradish committed:
kdeconnect: update to 21.12.3

eatradish committed:
sddm: use systemd-sysuser/tmpfiles to replace manual user/daemon directory creation

Co-authored-by: Mingcong Bai <jeffbai@aosc.xyz>

eatradish committed:
vtk: rebuild for Qt 5.15 KDE fork

eatradish committed:
openimageio: rebuild for Qt 5.15 KDE fork

eatradish committed:
multimc5: rebuild for Qt 5.15 KDE fork

MingcongBai committed:
citra: move prepare to correct location

MingcongBai committed:
melonds: use Clang on arm64

- GCC 11 does not support some C++ directives on arm64.

Co-authored-by: liushuyu <liushuyu011@gmail.com>

MingcongBai committed:
kvantum: fix build with KWindowSystem 5.91.0

MingcongBai committed:
liborcus: update checksum

MingcongBai committed:
pdfium: fix CHKSUMS array (remove one SKIP); symlink AArch64 toolchain

MingcongBai committed:
megasync: limit to amd64

MingcongBai committed:
lazarus: limit to amd64, i486

- Fails to build on AArch64 due to Glibc symbol incompatibility, a packaging
  fault in fpc (FreePascal) could have caused this issue. In the interest of
  time (this topic has dragged on for way too long), going for FAIL_ARCH.

MingcongBai committed:
rstudio: limit to amd64

MingcongBai committed:
libreoffice: add llvm build dep

MingcongBai committed:
groups/kde-survey-20220209-rebuilds: build gr-iqbal before gqrx for epoch fix

MingcongBai committed:
groups/kde-survey-20220209: drop kdgantt2

MingcongBai committed:
neovim-qt: limit to amd64, arm64

MingcongBai committed:
citra: drop old ppc64el patch for crypto++

MingcongBai committed:
goldendict: add ffmpeg dep

MingcongBai committed:
plasma-angelfish: limit to amd64, arm64 due to QtWebEngine dependency

MingcongBai committed:
sink: add gpgme dep

MingcongBai committed:
babe: limit to amd64, arm64 due to QtWebEngine dependency

MingcongBai committed:
makemkv: limit to amd64

MingcongBai committed:
firebird-emu: fix build on JIT-less architectures

MingcongBai committed:
multimc5: fix build with Java >= 11

Co-authored-by: eatradish <sakiiily@aosc.io>

MingcongBai committed:
rockbox-utility: drop broken architecture-platform conditionals in Makefiles

- These broken conditionals result in build failures on i486 and powerpc/ppc64/ppc64el.

MingcongBai committed:
lmms: fix build on ppc64el

- When using IBM long double, constexpr for float-point operations is unavailable in GCC

Reference: https://bugzilla.mozilla.org/show_bug.cgi?id=1444251

Co-authored-by: liushuyu <liushuyu011@gmail.com>

MingcongBai commented:
Note, skipping multiple packages for secondary architectures in interest of time. Please exclude from `groups/kde-survey-20220209-rebuilds` at build time.

- Loongson3: 
    - Dolphin Emulator (`dolphin-emu`) due to linker failure (hangs at linking stage, killing container).
    - PDFium (`pdfium`) due to Clang target detection failure (wait for LLVM 14).
    - Stellarium (`stellarium`) due to linker failure (could not find variable specification at offset 0x176e6).
- POWER: 
    - LibreOffice (`libreoffice`) due to Clang target detection (wait for LLVM 14).

MingcongBai committed:
qtcreator: use USTC mirror to work around user agent detection

- Official mirror throws 403 when downloading sources using wget.

MingcongBai committed:
spim: improve and fix build

- Fix build on loongson3 by undefining `mips'.
- Drop broken help/qtspim.qhc installation routine using a patch (previously
  done with a sed, which no longer functions).
- Enable parallelism for QtSpim build.
- Pass Autobuild3 flags at build-time.

MingcongBai committed:
dolphin-emu: disable LTO for loongson3 via CMake

- FIXME: This still results in stalls during final linkage.

MingcongBai committed:
digikam: set loongson3 as __NOWEBENGINE arch

MingcongBai committed:
pdfium: symlink loongson3 (mips64el) toolchain executables

- FIXME: pdfium still fails to build with a signal 11 during final linkage.

MingcongBai committed:
pyside2: use USTC mirror to work around user agent detection

- Official mirror throws 403 when downloading sources using wget.

MingcongBai committed:
wayland-protocols: use golden linker to workaround linkage failure for tests on loongson3

MingcongBai committed:
vc: disable stripping to fix linkage

Co-authored-by: liushuyu <liushuyu011@gmail.com>

MingcongBai committed:
yuzu: drop glibc 2.34 patch

Co-authored-by: liushuyu <liushuyu011@gmail.com>

MingcongBai committed:
polkit-qt-1: update to 0.114.0

This update contains a fix that suppresses a recurring error thrown by kded:

QDBusArgument: read from a write-only object

Reference: https://invent.kde.org/libraries/polkit-qt-1/-/commit/fb1667ac9a5437b4784d2806a3e816cbdbe404a5

MingcongBai committed:
liborcus: update checksum

MingcongBai committed:
rstudio: fix build

- Use OpenJDK 8 to compile build scripts.
- Install Node.js dependencies beforehand to avoid build issues.
- FIXME: RStudio still does not work due to a fault in QtWebEngine detection.

Co-authored-by: liushuyu <liushuyu011@gmail.com>

MingcongBai committed:
signon-kwallet-extension: move /usr/lib64 => /usr/lib, where applicable
